crates/matrix-sdk-ui/tests/integration/timeline/replies.rs

```diff
@@ -906,6 +908,106 @@ async fn test_send_reply_with_event_id() {
+
+    let timeline = room.timeline().await.unwrap();
+    let (_, mut timeline_stream) =
+        timeline.subscribe_filter_map(|item| item.as_event().cloned()).await;
+
+    let event_id_from_bob = event_id!("$event_from_bob");
+    let f = EventFactory::new();
+    server
+        .sync_room(
+            &client,
+            JoinedRoomBuilder::new(room_id).add_timeline_event(
+                f.text_msg("Hello from Bob").sender(&BOB).event_id(event_id_from_bob),
+            ),
+        )
+        .await;
+
+    let event_from_bob =
+        assert_next_matches!(timeline_stream, VectorDiff::PushBack { value } => value);
+
+    // Clear the timeline to make sure the old item does not need to be
+    // available in it for the reply to work.
+    timeline.clear().await;
+    assert_next_matches!(timeline_stream, VectorDiff::Clear);
+
+    // Now, let's reply to a message sent by `BOB`.
+    Mock::given(method("PUT"))
+        .and(path_regex(r"^/_matrix/client/v3/rooms/.*/send/.*"))
+        .respond_with(move |req: &Request| {
+            use ruma::events::room::message::RoomMessageEventContent;
+
+            let reply_event = req
+                .body_json::<RoomMessageEventContent>()
+                .expect("Failed to deserialize the event");
+
+            assert_matches!(reply_event.relates_to, Some(Relation::Thread(thread)) => {
+                assert_eq!(thread.event_id, event_id_from_bob);
+                assert_eq!(thread.in_reply_to.unwrap().event_id, event_id_from_bob);
+            });
+            assert_matches!(reply_event.mentions, None);
+
+            ResponseTemplate::new(200).set_body_json(json!({ "event_id": "$reply_event" }))
+        })
+        .expect(1)
+        .mount(server.server())
+        .await;
+
+    let replied_to_info = event_from_bob.replied_to_info().unwrap();
+    timeline
+        .send_thread_reply(
+            RoomMessageEventContentWithoutRelation::text_plain("Replying to Bob"),
+            replied_to_info,
+            ReplyWithinThread::No,
+        )
+        .await
+        .unwrap();
+
+    // Let the send queue handle the event.
+    yield_now().await;
+
+    let reply_item = assert_next_matches!(timeline_stream, VectorDiff::PushBack { value } => value);
+
+    assert_matches!(reply_item.send_state(), Some(EventSendState::NotSentYet));
+    let reply_message = reply_item.content().as_message().unwrap();
+    assert_eq!(reply_message.body(), "Replying to Bob");
+    let in_reply_to = reply_message.in_reply_to().unwrap();
+    assert_eq!(in_reply_to.event_id, event_id_from_bob);
+
+    // Right now, we don't pass along the replied-to event to the event handler,
+    // so it's not available if the timeline got cleared. Not critical, but
+    // there's notable room for improvement here.
+    //
+    // let replied_to_event =
+    // assert_matches!(&in_reply_to.event, TimelineDetails::Ready(ev) => ev);
+    // assert_eq!(replied_to_event.sender(), *BOB);
+
```

Comment:
Let's remove this, as it's copy pasta from the test that was the inspiration here :)